DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_default_configuration.py

```diff
@@ -0,0 +1,197 @@
+            "src_mac": src_mac,
+            "dst_mac": "ff:ff:ff:ff:ff:ff",
+            "vlanID": vlan
+        } for vlan in packet_vids if vlan != "X"}
+
+        streams.update({"Untagged traffic": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "src_mac": src_mac,
+            "dst_mac": "ff:ff:ff:ff:ff:ff"
+        }})
+    elif traffic_type == "multicast":
+        streams = {f"Traffic with VLAN ID: {vlan}": {
+            "type": "raw",
+            "rate": 50,
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "src_mac": src_mac,
+            "dst_mac": "01:80:C2:00:00:00",
+            "vlanID": vlan
+        } for vlan in packet_vids if vlan != "X"}
+
+        streams.update({"Untagged traffic": {
+            "type": "raw",
+            "rate": 50,
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "src_mac": src_mac,
+            "dst_mac": "01:80:C2:00:00:00"
+        }})
+    elif traffic_type == "unknown_unicast":
+        streams = {f"{tg_ports[0]} -> with VLAN: {vlan}": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "srcMac": src_mac,
+            "dstMac": "02:00:00:00:00:02",
+            "vlanID": vlan
+        } for vlan in packet_vids if vlan != "X"}
+
+        # Untagged stream
+        streams.update({"Untagged stream": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "srcMac": src_mac,
+            "dstMac": "02:00:00:00:00:02",
+        }})
+    else:
+        dst_mac = "02:00:00:00:00:04"
+        streams = {f"{tg_ports[0]} -> {vlan}": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": dev_groups[tg_ports[0]][0]["name"],
+            "ip_destination": [dev_groups[tg_ports[3]][0]["name"]],
+            "src_mac": src_mac,
+            "dst_mac": dst_mac,
+            "vlanID": vlan
+        } for vlan in packet_vids if vlan != "X"}
+
+        streams.update({f"{tg_ports[0]} -> X": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": dev_groups[tg_ports[0]][0]["name"],
+            "ip_destination": [dev_groups[tg_ports[3]][0]["name"]],
+            "src_mac": src_mac,
+            "dst_mac": dst_mac
+        }})
+
+        streams.update({f'{tg_ports[3]} -> {tg_ports[0]}': {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": dev_groups[tg_ports[3]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[0]][0]["name"],
+            "src_mac": dst_mac,
+            "dst_mac": src_mac,
+            "vlanID": 1
+        }})
```

Comment:
Convert it to a function that configures streams based on traffic type. There is much code duplication here.